File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i,k = len(nums)-1,len(nums)-1

        while i>=0:
            if nums[i] == val:
                nums[k],nums[i] = nums[i], nums[k]
                k-=1
            i-=1
        return k+1

File: Data Structures & Algorithms/rotate-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {void} Do not return anything, modify nums in-place instead.
     */
    rotate(nums, k) {
        let n = nums.length
        k = k%n
        nums.reverse()
        let start = 0, end = k-1
        while(start<end){
            let temp = nums[start]
            nums[start] = nums[end]
            nums[end]= temp
            start++
            end--
        }
        start = k,end = n-1
        while(start<end){
            let temp = nums[start]
            nums[start] = nums[end]
            nums[end]= temp
            start++
            end--
        }
    }
}
